plugins/hls-hlint-plugin/hls-hlint-plugin.cabal

```diff
@@ -54,14 +60,28 @@ library
     , transformers
     , unordered-containers
 
-  if ((!flag(ghc-lib) && impl(ghc >=8.10.1)) && impl(ghc <9.0.0))
-    build-depends: ghc ^>=8.10
+  if (flag(hlint33))
+    build-depends: hlint ^>=3.3
+  else
+    build-depends: hlint ^>=3.2
+
+  if (!flag(ghc-lib) && (!flag(hlint33) && impl(ghc >=8.10.1) || flag(hlint33) && impl(ghc >=9.0.1)) && impl(ghc <9.1.0))
```

Comment:
this condition is quite hard to follow, i wonder if it could be simplified

Reply:
Probably. This was just the first/most direct way I could come up with that worked.
I wanted to try to mirror the behaviour of hlint as closely as possible, so it's based on hlint's cabal file, but for two different versions of hlint at the same time: https://github.com/ndmitchell/hlint/commit/e2c0833c4a337f78f2cee58ef59d318d26f1666b#diff-3ab702ffba9192c29e2d49f82e65c36af604ea811daed8644bf3037fb6f4fc11.

Reply:
How about this version? It's a lot more direct and mirrors the intent of supporting two hlint versions more clearly, but has (very slightly) more code duplication
```
  if (flag(hlint33))
    build-depends: hlint ^>=3.3
    if (!flag(ghc-lib) && impl(ghc >=9.0.1) && impl(ghc <9.1.0))
      build-depends: ghc ==9.0.*

    else
        build-depends:
          , ghc
          , ghc-lib            == 9.0.*
          , ghc-lib-parser-ex  == 9.0.*

      cpp-options:   -DHLINT_ON_GHC_LIB

  else
    build-depends: hlint ^>=3.2
    if (!flag(ghc-lib) && impl(ghc >=8.10.1) && impl(ghc < 8.11.0))
        build-depends: ghc >=8.10 && < 9.0

    else
        build-depends:
          , ghc
          , ghc-lib            ^>= 8.10.4.20210206
          , ghc-lib-parser-ex  ^>= 8.10

      cpp-options:   -DHLINT_ON_GHC_LIB
```

Reply:
> How about this version? It's a lot more direct and mirrors the intent of supporting two hlint versions more clearly, but has (very slightly) more code duplication

I like it, it is more readable so the slight code duplication worths it imo